doc: Improve TLSStore CRD documentation (#9579)
traefik/traefik: docs/content/routing/providers/kubernetes-crd.md

`TLSStore` is the CRD implementation of a [Traefik "TLS Store"](../../https/tls.md#certificates-stores).

Register the `TLSStore` kind in the Kubernetes cluster before creating `TLSStore` objects.

!!! important "Default TLS Store"

    Traefik currently only uses the [TLS Store named "default"](../../https/tls.md#certificates-stores).

>     This _default_ `TLSStore` should be in Traefik's namespace. It is used by default on [`IngressRoute`](#kind-ingressroute) and [`IngressRouteTCP`](#kind-ingressroutetcp) objects.

Reviewer: ```suggestion
    This _default_ `TLSStore` should be in a namespace discoverable by Traefik. It is used by default on [`IngressRoute`](#kind-ingressroute) and [`IngressRouteTCP`](#kind-ingressroutetcp) objects.
```

>     This _default_ `TLSStore` should be in a namespace discoverable by Traefik. It is used by default on [`IngressRoute`](#kind-ingressroute) and [`IngressRouteTCP`](#kind-ingressroutetcp) objects.

Reviewer 2: ```suggestion
    This _default_ `TLSStore` should be in a namespace discoverable by Traefik. Since it is used by default on [`IngressRoute`](#kind-ingressroute) and [`IngressRouteTCP`](#kind-ingressroutetcp) objects, there never is a need to actually reference it.
```

    This means that you cannot have two stores that are named default in different Kubernetes namespaces.

>     For the time being, please only configure one TLSStore named default in Traefik's namespace.

Reviewer: ```suggestion
    For the time being, please only configure one TLSStore named default.
```

`TLSStore` is the CRD implementation of a [Traefik "TLS Store"](../../https/tls.md#certificates-stores).

Register the `TLSStore` kind in the Kubernetes cluster before creating `TLSStore` objects.

!!! important "Default TLS Store"

    Traefik currently only uses the [TLS Store named "default"](../../https/tls.md#certificates-stores).

>     This _default_ `TLSStore` should be in Traefik's namespace. It is used by
>     default on [`IngressRoute`](#kind-ingressroute) and
>     [`IngressRouteTCP`](#kind-ingressroutetcp) objects.

Reviewer 3: ```suggestion
    This _default_ `TLSStore` should be in Traefik's namespace. It is used by default on [`IngressRoute`](#kind-ingressroute) and [`IngressRouteTCP`](#kind-ingressroutetcp) objects.
```

    This means that you cannot have two stores that are named default in different Kubernetes namespaces.

>     For the time being, please only configure one TLSStore named default in Traefik namespace.

Reviewer 3: ```suggestion
    For the time being, please only configure one TLSStore named default in Traefik's namespace.
```

`TLSStore` is the CRD implementation of a [Traefik "TLS Store"](../../https/tls.md#certificates-stores).

Register the `TLSStore` kind in the Kubernetes cluster before creating `TLSStore` objects

> or referencing TLS stores in the [`IngressRoute`](#kind-ingressroute) / [`IngressRouteTCP`](#kind-ingressroutetcp) objects.

Reviewer: I think this sentence introduces what the section is about. It would still be accurate to say that it's about referencing a TLS store.

Author: I'm not sure to follow you, sorry, maybe I missed something. 
If we can only use one _default_ TLSStore, why a user would be interested in referencing a TLSStore on an IngressRoute ?

Reviewer: Yes, sorry the selection targeted was too large for my comment to make sense.
I think that the second part of the sentence (`referencing TLS stores in ....`), was to introduce what the section is about. It is the same wording as in other sections for other resources on this page.
The main issue with that part of the sentence is that it is only allowed to reference the "default" `tlsStore`, and hence we are here documenting something that does not need to be configured/referenced (as the "default" `tlsStore` is always used anyway).

> The _default_ `TLSStore` is used by default on [`IngressRoute`](#kind-ingressroute) and [`IngressRouteTCP`](#kind-ingressroutetcp) objects.

Reviewer: What do you think about moving this sentence to the note about the default TLS Store just below?

Author: Okay, let's do that.

    kind: TLSStore
    metadata:
      name: default

>       namespace: traefik

Reviewer: Why do you think we should change the namespace? I would remove it at most, but the `traefik` namespace is not shared by the resources below.

Reviewer: Yes 👍

    
    spec:
      entryPoints:

>         - websecure

Reviewer: Indeed, it would make more sense to attach to the `websecure` entryPoint, or more demonstrative. But it is not required, as nothing prevents attaching a TLS router on any port. I'm just wondering whether it was done on purpose.

Author: Me too :). If it was done on purpose, then maybe an explanation would help.

        - websecure
      routes:
      - match: Host(`example.com`) && PathPrefix(`/stripit`)
        kind: Rule
        services:
        - name: whoami
          port: 80

>       tls:

Reviewer: The TLS section should at least be empty to enable TLS.

Author: Got It. Updated in my last commit.

    
    spec:
      entryPoints:
        - websecure
      routes:
      - match: Host(`example.com`) && PathPrefix(`/stripit`)
        kind: Rule
        services:
        - name: whoami
          port: 80

>       tls:

Reviewer: ```suggestion
      tls: {}
```

Author: Yes, it's better like this. Thanks !
